src/screens/courses/CourseProfile/index.tsx

```diff
@@ -47,12 +48,38 @@ interface CourseProfileProps {
   resetCourseReducer: () => void,
 }
 
+const renderStepCell = ({ item, index }, course, route) => {
+  if (item.type === ON_SITE) {
+    return <OnSiteCell step={item} slots={course?.slots} index={index} profileId={route.params.courseId} />;
+  }
+
+  if (item.type === E_LEARNING) {
+    return <ELearningCell step={item} index={index} profileId={route.params.courseId}
+      endedActivity={route.params.endedActivity} />;
+  }
+
+  return null;
+};
+
+const renderSeparator = () => <View style={styles.separator} />;
```

Comment:
Je me demandais si le 3ème point de l'article pouvait être appliqué ici, mais je me demande si ce n'est pas moins compréhensible

Reply:
https://betterprogramming.pub/21-best-practices-for-a-clean-react-project-df788a682fb

Reply:
Je me suis posée la question également et j'ai statué sur non car les props ne sont pas les mêmes pour les 2 composants